beetmoverscript/task.py

```diff
@@ -134,6 +138,27 @@ def get_upstream_artifacts(context):
     return artifacts
 
 
+def get_release_props(context, platform_mapping=STAGE_PLATFORM_MAP):
+    """determined via parsing the Nightly build job's balrog_props.json and
+    expanded the properties with props beetmover knows about."""
+    payload_properties = context.task.get('payload', {}).get('release_properties', None)
+    initial_release_props_file = None
+
+    if payload_properties:
+        props = payload_properties
+        log.debug("Loading release_props from task's payload: {}".format(props))
+    else:
+        initial_release_props_file = get_initial_release_props_file(context)
+        props = utils.load_json(initial_release_props_file)['properties']
+        log.warn(
+            'Deprecated behavior! This will be gone after Firefox 59 reaches release. Loading release_props from "{}": {}'
+            .format(initial_release_props_file, props)
+        )
+
+    final_props = update_props(context, props, platform_mapping)
```

Comment:
This call should be called only in the `else` statement above. 
The `update_props()` was a hack we added at some point to map correctly some fields like: `platform`, `stage_platform` and alike. To be honest, I'm hoping all this goes away in the new implementation as we can nicely and easily set those up in-tree. Now that we have `shipping_phase` and `shipping_product` we can infer all these previously-hacked values via transforms in-tree. So we don't need to perform all these hacks within beetmoverscript.

For context, the reason why all these hacks are here, is because the `balrog_props` file, generated at build time, doesn't look the same across products and platforms. It's missing `stage_platform` in some products, but is present in others. Likewise for `platform` or `locale`. But in-tree, we can finally standardize and append all these fields so that we can finally clean this up.

Reply:
That's good to know. Do you want me to make the changes in this PR or a follow would be good enough?

Reply:
Either wfm, we can address it later on too, just not to forget of it.

Reply:
Okay, I'll go with a follow up. I'm not familiar enough with beetmover to be sure it's okay to remove these pieces of data

Reply:
Follow up at #104